catalog/content/core/content-core-videothumbnailplugin/src/main/java/ddf/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.content.plugin.video;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.concurrent.Semaphore;
+
+import javax.imageio.ImageIO;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecuteResultHandler;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.Executor;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.SystemUtils;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.content.data.ContentItem;
+import ddf.content.operation.CreateRequest;
+import ddf.content.operation.CreateResponse;
+import ddf.content.operation.UpdateRequest;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.operation.impl.CreateResponseImpl;
+import ddf.content.operation.impl.UpdateResponseImpl;
+import ddf.content.plugin.PluginExecutionException;
+import ddf.content.plugin.PostCreateStoragePlugin;
+import ddf.content.plugin.PostUpdateStoragePlugin;
+
+public class VideoThumbnailPlugin implements PostCreateStoragePlugin, PostUpdateStoragePlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VideoThumbnailPlugin.class);
+
+    private static final int MAX_FFMPEG_PROCESSES = 4;
```

Comment:
It's an educated guess based on the resource usage I observed while running FFmpeg. It certainly wouldn't hurt anything to make it configurable, but I doubt anyone would bother to change it.